integrations/mongodb_atlas/tests/test_document_store.py

```diff
@@ -192,7 +192,168 @@ def test_custom_embedding_field(self):
+
+    @pytest.mark.integration
+    def test_document_conversion_methods(self):
+        """Test the document conversion helper methods with custom field mappings."""
+        database_name = "haystack_integration_test"
+        collection_name = "test_document_conversion_" + str(uuid4())
+
+        connection: MongoClient = MongoClient(
+            os.environ["MONGO_CONNECTION_STRING"], driver=DriverInfo(name="MongoDBAtlasHaystackIntegration")
+        )
+        database = connection[database_name]
+        if collection_name in database.list_collection_names():
+            database[collection_name].drop()
+        database.create_collection(collection_name)
+        database[collection_name].create_index("id", unique=True)
+
+        try:
+            # Create a document store with custom field names
+            custom_store = MongoDBAtlasDocumentStore(
+                database_name=database_name,
+                collection_name=collection_name,
+                vector_search_index="cosine_index",
+                full_text_search_index="full_text_index",
+                embedding_field="custom_vector",
+                content_field="custom_text",
+            )
+
+            # Test haystack_doc_to_mongo_doc
+            haystack_doc = Document(content="test content", embedding=[0.1, 0.2, 0.3], meta={"test_meta": "test_value"})
+
+            mongo_doc = custom_store._haystack_doc_to_mongo_doc(haystack_doc)
+
+            # Check field mapping
+            assert "custom_text" in mongo_doc
+            assert mongo_doc["custom_text"] == "test content"
+            assert "content" not in mongo_doc
+
+            assert "custom_vector" in mongo_doc
+            assert mongo_doc["custom_vector"] == [0.1, 0.2, 0.3]
+            assert "embedding" not in mongo_doc
+
+            assert "meta" in mongo_doc
+            assert mongo_doc["meta"] == {"test_meta": "test_value"}
+
+            # Test mongo_doc_to_haystack_doc
+            converted_doc = {
+                "id": "test_id",
+                "custom_text": "test content from mongo",
+                "custom_vector": [0.4, 0.5, 0.6],
+                "meta": {"mongo_meta": "mongo_value"},
+                "_id": "mongodb_internal_id",  # This should be removed
+            }
+
+            haystack_doc = custom_store._mongo_doc_to_haystack_doc(converted_doc)
+
+            # Check field mapping back to haystack format
+            assert haystack_doc.content == "test content from mongo"
+            assert haystack_doc.embedding == [0.4, 0.5, 0.6]
+            assert haystack_doc.meta == {"mongo_meta": "mongo_value"}
+            assert haystack_doc.id == "test_id"
+
+            # Make sure MongoDB's internal _id is removed
+            assert not hasattr(haystack_doc, "_id")
+
+        finally:
+            database[collection_name].drop()
+
+    @pytest.mark.integration
+    def test_document_conversion_with_unsupported_fields(self):
```

Comment:
I would transform this integration test into a unit test

Reply:
fair